tests/integration/modifyAccount.test.ts

```diff
@@ -1,116 +1,233 @@
+      password: "password",
+      confirmedPassword: "password",
+      orgCategory: "Other" as OrgCategory
+    }
+
+    const infoUser = await createUserWithEmailAndPassword(
+      auth,
+      info.email,
+      info.password
     )
+
+    expect(infoUser.user).toBeDefined()
+  })
+
+  it("can tell if a user is logging in", async () => {
+    await signOut(auth)
+    const user = await createUser()
+
+    expectCurrentUser()
+
+    user.email && (await signInUser(user.email))
+
+    expectCurrentUser(user)
+
+    await signOut(auth)
+
+    expectCurrentUser()
   })
 
-  return creds
-}
+  it("sets a profile", async () => {
+    await signInTestAdmin()
+    const userInfo = { ...fakeUser(), role: "user" as Role }
+    await setProfile(userInfo.uid, userInfo)
+    const profile = await getDoc(doc(firestore, `/profiles/${userInfo.uid}`))
+    expect(profile.data()).toMatchObject(userInfo)
+    await signOut(auth)
+  })
+})
 
 describe("modifyAccount", () => {
-  it("finishSignup creates pendingUpgrade accounts", async () => {
-    const creds = await createPendingUpgradeUser()
+  it("runs", async () => {
+    const user = await createUser()
+    expect(user?.email).toBeDefined()
 
-    expect((await creds.user.getIdTokenResult(true)).claims).toMatchObject({
+    const adminUser = await signInTestAdmin()
+    const admin = await testAuth.getUser(adminUser.uid)
+    expect(admin).toBeDefined()
+
+    const role = "user"
+    const isPublic = true
+    expect(user).toBeDefined()
+    await expectUser(admin, "admin", false)
+
+    await modifyAccount({ uid: user.uid, role })
+      .then(async () => await expectUser(user, role, isPublic))
+      .catch(async err => console.log({ err }))
+
+    await modifyAccount({ uid: user.uid, role: "organization" })
+
+    await expectUser(user, "organization" as Role, isPublic)
+
+    await modifyAccount({ uid: user.uid, role: "admin" })
+
+    await expectUser(user, "admin" as Role, isPublic)
+
+    await signOut(auth)
+  })
+})
+
+describe("finishSignup_create new user", () => {
+  it("finishSignup callable assigns organization request to role pendingupgrade", async () => {
+    const user = await createUser()
+
+    const isPublic = true
+    expect(user).toBeDefined()
+
+    await signInUser(user.email!)
+
+    await finishSignup({ requestedRole: "organization" })
+
+    const updated = (await auth.currentUser?.getIdTokenResult(true))?.claims
+    expect(updated?.role).toEqual("pendingUpgrade")
+
+    await signOut(auth)
+    expectCurrentUser()
+    await signInUser(user.email!)
+    expectCurrentUser(user)
+
+    await act(async () => await deleteUser(auth.currentUser!))
+  })
+
+  it("create user with email hook creates pendingUpgrade accounts", async () => {
+    const newUser = {
+      email: `${nanoid(6)}@example.com`,
+      password: "password",
+      confirmedPassword: "password",
+      fullName: `Test ${nanoid(4)}`
+    } as CreateUserWithEmailAndPasswordData
```

Comment:
```suggestion
    const newUser: CreateUserWithEmailAndPasswordData = {
      email: `${nanoid(6)}@example.com`,
      password: "password",
      confirmedPassword: "password",
      fullName: `Test ${nanoid(4)}`
    }
```

Reply:
Generally avoiding `as` is a good practice, and if you have to use it (or want to use it) create a comment right above the usage like:
```ts
// Type Safety: this is safe because ...
```

The reason to avoid it is that it removes some type safety that the compiler gives us. For example, if we change the structure of `newUser` when it's being created, it will still be considered a `CreateUserWithEmailAndPasswordData` type, yet not conform to that particular type.